lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -96,15 +96,27 @@ sub list {
         }
         if ($have) {
             for my $par (@$key) {
-                $search{$par} = $self->param($par);
+                $search{"me.$par"} = $self->param($par);
             }
         }
     }
 
     my @result;
+    my $sql_field = {
+        Machines => 'machine',
+        TestSuites => 'test_suite',
+        Products => 'product',
+    }->{$table};
```

Comment:
I'm wondering whether we can ask DBIx for the raw table name, e.g. maybe https://metacpan.org/pod/DBIx::Class::ResultSource#name would return it. That would be better than hardcoding the mapping here. Otherwise, it could also be done via a generic CamlCase to snake_case conversion.